Require `puppet_x/helper.rb` based on full path

sigv opened the pull request:
Not all code ends up in Puppet Ruby LOAD_PATH. We are running a fleet of Puppet 6.20.0 and Puppet 7.3.0 nodes, which cannot autoload `puppet/provider/package/windowspowershell` as there is no such file to load -- `puppet_x/encore/powershellmodule/helper`.

Using a dynamically expanded path from the `lib/puppet/` files seems to very much so do the trick. There was a 2012 O'Reilly book [Puppet Types and Providers (Chapter 4: Section "Shared Libraries")](https://www.oreilly.com/library/view/puppet-types-and/9781449339319/ch04.html) talking about this as well. Might not be the newest and greatest solution, as it would be possible to work with LOAD_PATH, but this solves the underlying catalog generation error.

nmaludy reviewed (changes requested):
@sigv thanks for this fix, can you please add a change log entry and ill get it merged!

sigv committed:
Require `puppet_x/helper.rb` based on full path

Not all code ends up in Puppet Ruby LOAD_PATH. We are running
a fleet of Puppet 6.20.0 and Puppet 7.3.0 nodes, which cannot
autoload `puppet/provider/package/windowspowershell` as there
is no such file to load -- `puppet_x/encore/powershellmodule/helper`.

Using a dynamically expanded path from the `lib/puppet/` files
seems to very much so do the trick. There was a 2012 O'Reilly
book talking about this as well. Might not be the newest and greatest
solution, as it would be possible to work with LOAD_PATH, but this
solves the underlying catalog generation error.

Book: Puppet Types and Providers: Chapter 4: Shared Libraries
> https://www.oreilly.com/library/view/puppet-types-and/9781449339319/ch04.html

sigv commented:
Thank you for the fast approval!
What does your standard release schedule look like?
(When could a patch release be expected on Forge?)

sigv commented:
Oh, `2.2.1` is already there! 😮 👌🏻 
Thank you very much then.

nmaludy commented:
@sigv you're welcome, enjoy!
